src/sinks/socket.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    sinks::util::tcp::{Encoding, TcpSinkConfig, TlsConfig},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+// TODO: add back when serde-rs/serde#1358 is addressed
+// #[serde(deny_unknown_fields)]
+pub struct SocketSinkConfig {
+    #[serde(flatten)]
+    pub mode: Mode,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(tag = "mode", rename_all = "snake_case")]
+pub enum Mode {
+    Tcp(TcpSinkConfig),
+}
```

Comment:
I think it is worth writing an `impl From<TcpSinkConfig> for Mode`, which would eliminate the need for `Mode::Tcp(TcpSinkConfig)` elsewhere, and allow instead `TcpSinkConfig.into()`

Reply:
Done!